Source/hydro/fourth_order.F90

```diff
@@ -448,24 +448,30 @@ subroutine ca_make_cell_center(lo, hi, &
     endif
 
     do n = 1, nc
-       do k = lo(3), hi(3)
-          do j = lo(2), hi(2)
-             do i = lo(1), hi(1)
 
-                lap = U(i+1,j,k,n) - TWO*U(i,j,k,n) + U(i-1,j,k,n)
+       if (maxval(abs(U(U_lo(1)-1:U_hi(1)+1, U_lo(2)-dg(2):U_hi(2)+dg(2), U_lo(3)-dg(3):U_hi(3)+dg(3), n))) == ZERO) then
```

Comment:
Is this change for performance or correctness? If it's performance, I'm not sure it's actually worth it, I suspect we won't save that much since we'll still have to pull in all of the data from memory and there's only ~3 flops/byte in this conversion, so this calculation is probably always memory bandwidth bound.

Reply:
the idea was performance, but I had the same concern as you.  I'll revert that one.

Reply:
I removed them all -- tests showed no performance difference